### PR TITLE
Use Go 1.18

### PR DESCRIPTION
**Pull request by epk**
Signed-off-by: Aditya Sharma <git@adi.run>

### Description of your changes

Opened in place of #209 

- Use Go 1.18
- Fix misc lint errors

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

`make reviewable` + full CI run

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,11 +72,14 @@ build.artifacts.bundle.platform:
...
 
 build.artifacts.pkg.platform:
-	@cat nfpm_up.yaml | GO_BIN_DIR=$(GO_BIN_DIR) envsubst > $(CACHE_DIR)/nfpm_up.yaml
-	@cat nfpm_docker-credential-up.yaml | GO_BIN_DIR=$(GO_BIN_DIR) envsubst > $(CACHE_DIR)/nfpm_docker-credential-up.yaml
+	@mkdir -p $(CACHE_DIR)
 	@mkdir -p $(OUTPUT_DIR)/deb/$(PLATFORM)
-	@CACHE_DIR=$(CACHE_DIR) OUTPUT_DIR=$(OUTPUT_DIR) PLATFORM=$(PLATFORM) PACKAGER=deb $(GO) generate -tags packaging ./...
 	@mkdir -p $(OUTPUT_DIR)/rpm/$(PLATFORM)
+	@touch $(CACHE_DIR)/nfpm_up.yaml
+	@touch $(CACHE_DIR)/nfpm_docker-credential-up.yaml
```
Comment (hasheddan) [inline Makefile:79]: @epk were you getting errors if you didn't `touch` before piping `cat` into these files?

File: internal/xpls/server/server.go [hunk 2]
```diff
@@ -295,37 +295,37 @@ func (s *Server) registerWatchFilesCapability() {
...
+		s.publishDiagnostics(ctx, &protocol.PublishDiagnosticsParams{
 			URI:         protocol.URIFromSpanURI(uri),
 			Diagnostics: diags,
 		})
 	}()
 }
 
 // // watchSnapshot watches the cache for changes.
-func (s *Server) watchSnapshot() { // nolint:gocyclo
+func (s *Server) watchSnapshot(ctx context.Context) { // nolint:gocyclo
```
Comment (tnthornton) [inline internal/xpls/server/server.go:328]: @epk do we need to share the top level context from the server down in these background tasks? (this question covers this change as well as the other ctx passes)

File: Makefile (hunk 1 above)
Comment (epk) [reply Makefile]: It wasn't needed after all 😃

File: internal/xpls/server/server.go (hunk 2 above)
Comment (epk) [reply internal/xpls/server/server.go:328]: Thanks for catching this, I will fix forward in a separate PR, for now I added `//nolint:contextcheck` to keep the linter happy

**Review by tnthornton — APPROVED** [review]

**Review by hasheddan — APPROVED** [review]
Thanks @epk!

**Pull request merged**